### PR TITLE
update space_space keyboard

### PR DESCRIPTION
**Pull request by qpockets**
## Description
added rev2 of space space keyboard (new pinout & keymaps)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
There is a mix of tabs and spaces in your files.  Also, a lot of it doesn't follow the coding conventions.  

If you could, run `qmk cformat /keyboards/space_space/**` and commit the changes.  This should fox most of the formatting issues.

**Comment by qpockets** [comment]
Thanks Drashna. 
I've run the command and added a new commit

**Comment by qpockets** [comment]
could someone give this another look when you get a chance? 
it will be going up for sale very soon.

thanks!

File: keyboards/space_space/readme.md [hunk 1]
```diff
@@ -1,18 +1,18 @@
 # space_space
 
-![space_space](https://i.imgur.com/qdkG8K7l.jpg)
+![space_space](https://i.imgur.com/SxjhzIM.jpg)
```
Comment (drashna) [inline keyboards/space_space/readme.md:3]:
```suggestion
![space_space](https://i.imgur.com/SxjhzIMl.jpg)
```

File: keyboards/space_space/readme.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
 
 Give your space some space  
 
 * Keyboard Maintainer: https://github.com/qpockets
-* Hardware Availability: 
+* Hardware Availability: https://p3dstore.com/products/space-space-acrylic-case?_pos=21&_sid=c75de6a78&_ss=r&variant=39907740844216
```
Comment (drashna) [inline keyboards/space_space/readme.md:8]:
```suggestion
* Hardware Availability: [p3dstore](https://p3dstore.com/products/space-space-acrylic-case?_pos=21&_sid=c75de6a78&_ss=r&variant=39907740844216)
```

File: keyboards/space_space/rev2/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                }
+    } else if (index == 0) { /* right encoder */
+                if (clockwise){
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/space_space/rev2/keymaps/default/keymap.c:75]:
```suggestion
```

File: keyboards/space_space/rev2/keymaps/default/rules.mk [hunk 4]
```diff
@@ -0,0 +1 @@
+COMBO_ENABLE = yes
```
Comment (drashna) [inline keyboards/space_space/rev2/keymaps/default/rules.mk:1]: For the default keymap, combos shouldn't be enabled.

File: keyboards/space_space/rev2/keymaps/qpockets/keymap.c [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+                }
+    } else if (index == 0) { /* right encoder */
+                if (clockwise){
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/space_space/rev2/keymaps/qpockets/keymap.c:75]:
```suggestion
```

File: keyboards/space_space/rev2/keymaps/big_space/keymap.c [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+                }
+    } else if (index == 0) { /* right encoder */
+                if (clockwise){
+                    tap_code(KC_VOLU);
+                } else {
+                    tap_code(KC_VOLD);
+        }
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/space_space/rev2/keymaps/big_space/keymap.c:81]:
```suggestion
```

File: keyboards/space_space/rev2/rev2.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev2.h"
```
Comment (drashna) [inline keyboards/space_space/rev2/rev2.c:17]: This way, the encoder function is set by default but can be overridden by users, and will still work for the configurator.
```suggestion
#include "rev2.h"

bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 1) { /* left encoder*/
                if (clockwise){
                    tap_code(KC_WH_U);
                } else {
                    tap_code(KC_WH_D);
                }
    } else if (index == 0) { /* right encoder */
                if (clockwise){
                    tap_code(KC_VOLU);
                } else {
                    tap_code(KC_VOLD);
        }
    }
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/space_space/rev2/keymaps/default/rules.mk (hunk 4 above)
Comment (qpockets) [reply keyboards/space_space/rev2/keymaps/default/rules.mk]: commented out.
could have changed it to 'no' i guess, but i just thought of that 😄 
thanks for taking the time!

File: keyboards/space_space/rev1/keymaps/big_space/keymap.c [hunk 8]
```diff
@@ -52,18 +52,18 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
     KC_HOME, KC_UP,   KC_END,   KC_PGUP, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_BSPC,
     KC_LEFT, KC_DOWN, KC_RIGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TAB,
     KC_MPRV, KC_MPLY, KC_MNXT,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_VOLD, KC_VOLU, KC_ENT,
-                    KC_TRNS,   KC_TRNS,   KC_TRNS
+                    KC_TRNS,   KC_TRNS,   KC_TRNS 
   ),
 };
 
-bool encoder_update_user(uint8_t index, bool clockwise) {
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/big_space/keymap.c:66]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/space_space/rev1/keymaps/big_space/keymap.c [hunk 9]
```diff
@@ -112,7 +112,6 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
                 break;
         }
     }
-    return true;
 }
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/big_space/keymap.c:115]:
```suggestion
    return true;
}
```

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 10]
```diff
@@ -47,25 +47,25 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
     KC_HOME, KC_UP,   KC_END,   KC_PGUP, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_BSPC,
     KC_LEFT, KC_DOWN, KC_RIGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TAB,
     KC_MPRV, KC_MPLY, KC_MNXT,  KC_TRNS, KC_TRNS, KC_LCAP, KC_TRNS, KC_TRNS, KC_VOLD, KC_VOLU, KC_ENT,
-                    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS
+                    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS 
   ),
 };
 
-bool encoder_update_user(uint8_t index, bool clockwise) {
+void encoder_update_user(uint8_t index, bool clockwise) {
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:68]:
```suggestion
bool encoder_update_user(uint8_t index, bool clockwise) {
```

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 11]
```diff
@@ -96,7 +96,7 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
       }
     } else if (index == 0) { /* right encoder */
         switch(get_highest_layer(layer_state)){
-
+
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:99]:
```suggestion

```

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 12]
```diff
@@ -114,7 +114,6 @@ bool encoder_update_user(uint8_t index, bool clockwise) {
                 break;
         }
     }
-    return true;
 }
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:117]:
```suggestion
    return true;
}
```

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 13]
```diff
@@ -13,7 +13,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
+
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:16]:
```suggestion

```

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 14]
```diff
@@ -47,25 +47,25 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_TRNS, KC_6,    KC_7,    KC_8,    KC_9,     KC_0,
     KC_EXLM, KC_AT,   KC_HASH, KC_DLR,  KC_PERC, KC_TRNS, KC_CIRC, KC_AMPR, KC_ASTR, KC_EQUAL, KC_MINS,
     KC_PIPE, KC_BSLS, KC_LPRN, KC_LBRC, KC_SCLN, KC_TRNS, KC_COLN, KC_RBRC, KC_RPRN, KC_PLUS,  KC_UNDS,
-                    KC_TRNS,    KC_TRNS,   KC_TRNS,    KC_TRNS,      KC_TRNS,    KC_TRNS
+                    KC_TRNS,    KC_TRNS,   KC_TRNS,    KC_TRNS,      KC_TRNS,    KC_TRNS
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:50]:
```suggestion
                    KC_TRNS,    KC_TRNS,   KC_TRNS,    KC_TRNS,      KC_TRNS,    KC_TRNS
```

File: keyboards/space_space/rev1/keymaps/big_space/keymap.c [hunk 15]
```diff
@@ -13,7 +13,7 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
+
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/big_space/keymap.c:16]:
```suggestion

```

File: keyboards/space_space/rev2/keymaps/default/rules.mk [hunk 16]
```diff
@@ -0,0 +1 @@
+//COMBO_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/space_space/rev2/keymaps/default/rules.mk:1]:
```suggestion
COMBO_ENABLE = yes
```

File: keyboards/space_space/rev2/readme.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
+# space_space
+
+![space_space](https://i.imgur.com/SxjhzIM.jpg)
```
Comment (fauxpark) [inline keyboards/space_space/rev2/readme.md:3]:
```suggestion
![space_space](https://i.imgur.com/SxjhzIMh.jpg)
```

File: keyboards/space_space/rev2/rev2.h [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+    K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K50, \
+    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K51, \
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K52, \
+              K32, K33, K34,      K36, K37, K38 \
+) \
+{ \
+    {K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K50}, \
+    {K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K51}, \
+    {K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K52}, \
+    {XXX, XXX, K32, K33, K34, XXX, K36, K37, K38, XXX, XXX}, \
```
Comment (fauxpark) [inline keyboards/space_space/rev2/rev2.h:33]:
```suggestion
    {XXX, XXX, K32, K33, K34, XXX, K36, K37, K38, XXX, XXX} \
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by qpockets** [comment]
appreciate you taking a look fauxpark. these start going out at the end of the week so it's a big help to have qmk support, thank you

File: keyboards/space_space/rev1/keymaps/default/keymap.c [hunk 19]
```diff
@@ -61,7 +60,7 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
     KC_HOME, KC_UP,   KC_END,   KC_PGUP, KC_TRNS, RESET,   KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_BSPC,
     KC_LEFT, KC_DOWN, KC_RIGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TAB,
     KC_MPRV, KC_MPLY, KC_MNXT,  KC_TRNS, KC_TRNS, KC_LCAP, KC_TRNS, KC_TRNS, KC_VOLD, KC_VOLU, KC_ENT,
-                    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS
+                    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS
```
Comment (fauxpark) [inline keyboards/space_space/rev1/keymaps/default/keymap.c:63]:
```suggestion
                    KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS
```

File: keyboards/space_space/rev2/rev2.h [hunk 20]
```diff
@@ -0,0 +1,49 @@
...
+    K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K51, \
+    K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K52, \
+              K32,           K35,           K38 \
+) \
+{ \
+   {K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K50}, \
+   {K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K51}, \
+   {K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K52}, \
+   {XXX, XXX, K32, XXX, XXX, K35, XXX, XXX, K38, XXX, XXX} \
+} \
```
Comment (fauxpark) [inline keyboards/space_space/rev2/rev2.h:47]:
```suggestion
}
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**